lib/helpers.bash

```diff
@@ -92,980 +96,946 @@ alias reload_completion="$(_make_reload_alias completion completion)"
-  fi
-
-  echo "All done!"
-  echo ""
-  echo "Profile location: $profile_path"
-  echo "Load the profile by invoking \"bash-it profile load $name\""
+function _bash-it-profile-save() {
+	_about 'saves the current configuration to the "profile" directory'
+	_group 'lib'
+
+	local name="${1:-}"
+	while [[ -z "$name" ]]; do
+		read -r -e -p "Please enter the name of the profile to save: " name
+		case $name in
+			"")
+				echo -e "${echo_orange?}Please choose a name.${echo_reset_color?}"
+				;;
+			*)
+				break
+				;;
+		esac
+	done
+
+	local profile_path="${BASH_IT}/profiles/${name}.bash_it" RESP
+	if [[ -s "$profile_path" ]]; then
+		echo -e "${echo_yellow?}Profile \"$name\" already exists.${echo_reset_color?}"
+		while true; do
+			read -r -e -n 1 -p "Would you like to overwrite existing profile? [y/N] " RESP
+			case $RESP in
+				[yY])
+					echo -e "${echo_green?}Overwriting profile \"$name\"...${echo_reset_color?}"
+					rm "$profile_path"
+					break
+					;;
+				[nN] | "")
+					echo -e "${echo_orange?}Aborting profile save...${echo_reset_color?}"
+					return 1
+					;;
+				*)
+					echo -e "${echo_orange?}Please choose y or n.${echo_reset_color?}"
+					;;
+			esac
+		done
+	fi
+
+	local something_exists subdirectory
+	echo "# This file is auto generated by Bash-it. Do not edit manually!" > "$profile_path"
+	for subdirectory in "plugins" "completion" "aliases"; do
+		local component_exists="" f
+		echo "Saving $subdirectory configuration..."
+		for f in "${BASH_IT}/$subdirectory/available/"*.bash; do
+			_bash-it-determine-component-status-from-path "$f"
+			if [[ "$enabled" == "x" ]]; then
+				if [[ -z "$component_exists" ]]; then
+					# This is the first component of this type, print the header
+					component_exists="yes"
+					something_exists="yes"
+					echo "" >> "$profile_path"
+					echo "# $subdirectory" >> "$profile_path"
+				fi
+				echo "$subdirectory $enabled_file" >> "$profile_path"
+			fi
+		done
+	done
+	if [[ -z "$something_exists" ]]; then
+		echo "It seems like no configuration was enabled.."
+		echo "Make sure to double check that this is the wanted behavior."
+	fi
+
+	echo "All done!"
+	echo ""
+	echo "Profile location: $profile_path"
+	echo "Load the profile by invoking \"bash-it profile load $name\""
 }
 
 _bash-it-profile-load-parse-profile() {
-  _about 'Internal function used to parse the profile file'
-  _param '1: path to the profile file'
-  _param '2: dry run- only check integrity of the profile file'
-  _example '$ _bash-it-profile-load-parse-profile "profile.bash_it" "dry"'
-
-  local num=0
-  while read -r -a line; do
-    num=$((num + 1))
-    # Ignore comments and empty lines
-    [[ -z "${line[*]}" || "${line[*]}" =~ ^#.* ]] && continue
-    local enable_func="_enable-${line[0]}"
-    local subdirectory=${line[0]}
-    local component=${line[1]}
-
-    typeset to_enable=$(command ls "${BASH_IT}/$subdirectory/available/$component".*bash 2>/dev/null | head -1)
```

Comment:
@NoahGorny, I've seen this pattern all across this repo and it's a pretty strong [code smell]( https://en.m.wikipedia.org/wiki/Code_smell ). It's not *automatically* bad every time, but it's probably bad. 

99% of the time it just means that the variable shouldn't exist. Put the glob directly in the loop. The last 1% of the time, make the variable an array and stick the glob directly between the parentheses. 

(Aside: Completions code are weirding and operate rules perpendicular to normality. Aka, completions always smell!)